src/cargo/core/compiler/mod.rs

```diff
@@ -235,13 +235,24 @@ fn rustc(cx: &mut Context<'_, '_>, unit: &Unit, exec: &Arc<dyn Executor>) -> Car
     } else {
         format!("{}.d", unit.target.crate_name())
     };
+
     let rustc_dep_info_loc = root.join(dep_info_name);
     let dep_info_loc = fingerprint::dep_info_loc(cx, unit);
 
     rustc.args(cx.bcx.rustflags_args(unit));
     if cx.bcx.config.cli_unstable().binary_dep_depinfo {
         rustc.arg("-Z").arg("binary-dep-depinfo");
     }
+
+    if unit.target.get_binary_name().is_some() {
+        let mut filepath = String::from("--emit=link=");
+        let file = outputs
+            .iter()
+            .find(|output| output.flavor == FileFlavor::Normal)
+            .unwrap();
+        filepath.push_str(&file.path.to_string_lossy());
+    }
```

Comment:
I think this might be best to go into the `prepare_rustc` method where most other flags are configured if possible.

Reply:
We can get rid of this complete codeblock. I just tested the code without this, it still works.